src/TestFramework/Coverage/XmlReport/XmlCoverageParser.php

```diff
@@ -74,24 +74,25 @@ private static function createTestLocationsGenerator(SafeDOMXPath $xPath): itera
 
     private static function retrieveTestLocations(SafeDOMXPath $xPath): TestLocations
     {
-        $linesNode = $xPath->query('/p:phpunit/p:file/p:totals/p:lines')[0];
-
-        $percentage = $linesNode->getAttribute('percent');
+        // @phpstan-ignore method.nonObject
```

Comment:
here there is no guarantee that the percentage is found: should we leave it as is (which can cause a `TypeError`) or check for it?

Note that this is directly from reading the coverage report which is a hotpath: we can justify risking an obscure error in favour of performance.